### PR TITLE
[EXC-6558] - Adding customDomains property

### PR DESCRIPTION
**Pull request by marianomerlo**

### CODE REVIEW

File: mule-packager/src/main/java/org/mule/tools/api/util/MavenProjectInformation.java [hunk 1]
```diff
@@ -156,13 +164,14 @@ public Pom getEffectivePom() {
   }
 
   private static Optional<ExchangeRepositoryMetadata> getExchangeRepositoryMetadata(DeploymentRepository repository,
-                                                                                    Settings settings) {
+                                                                                    Settings settings,
+                                                                                    List<String> customDomains) {
     ExchangeRepositoryMetadata metadata = null;
-    if (repository != null && ExchangeRepositoryMetadata.isExchangeRepo(repository.getUrl())) {
+    if (repository != null && new ExchangeUriChecker(customDomains).isExchangeRepo(repository.getUrl())) {
```
Comment (Igua95) [inline mule-packager/src/main/java/org/mule/tools/api/util/MavenProjectInformation.java:170]: Could we initialize the `ExchangeUriChecker` instance in a different line, It seems too much for an if condition

File: mule-packager/src/main/java/org/mule/tools/api/util/MavenProjectInformation.java [hunk 2]
```diff
@@ -62,14 +70,14 @@ public static MavenProjectInformation getProjectInformation(MavenSession session
...
-      Optional<ExchangeRepositoryMetadata> metadata = getExchangeRepositoryMetadata(repository, settings);
+      Optional<ExchangeRepositoryMetadata> metadata = getExchangeRepositoryMetadata(repository, settings, customDomains);
       if (metadata.isPresent()) {
         if (deployments != null && isPlatformDeployment(deployments)) {
           builder.withDeployments(deployments);
         }
 
         // updating repository uri in order to deploy to Exchange
-        ExchangeRepositoryBuilder.updateRepositoryUri(project, session.getRequest().getSystemProperties());
+        new ExchangeHelper(customDomains).updateRepositoryUri(project, session.getRequest().getSystemProperties());
```
Comment (Igua95) [inline mule-packager/src/main/java/org/mule/tools/api/util/MavenProjectInformation.java:80]: Could we destructure it, to make it more readable?

ExchangeHelper exchangeHelper = new ExchangeHelper(customDomains);
var? systemProperties = session.getRequest().getSystemProperties();

and then just:
`exchangeHelper.updateRepositoryUri(project,systemProperties);`

File: mule-packager/src/main/java/org/mule/tools/api/util/MavenProjectInformation.java (hunk 1 above)
Comment (marianomerlo) [reply mule-packager/src/main/java/org/mule/tools/api/util/MavenProjectInformation.java:170]: If that makes you happy...

File: mule-packager/src/main/java/org/mule/tools/api/util/MavenProjectInformation.java [hunk 3]
```diff
@@ -167,7 +170,8 @@ public Pom getEffectivePom() {
                                                                                     Settings settings,
                                                                                     List<String> customDomains) {
     ExchangeRepositoryMetadata metadata = null;
-    if (repository != null && new ExchangeUriChecker(customDomains).isExchangeRepo(repository.getUrl())) {
+    ExchangeUriChecker exchangeUriChecker = new ExchangeUriChecker(customDomains);
+    if (repository != null &&exchangeUriChecker.isExchangeRepo(repository.getUrl())) {
```
Comment (Igua95) [inline mule-packager/src/main/java/org/mule/tools/api/util/MavenProjectInformation.java:174]: if you need to commit something else, place a space between & and the second clause

**Review by Igua95 — APPROVED** [review]
👍

File: mule-packager/src/main/java/org/mule/tools/api/util/MavenProjectInformation.java (hunk 3 above)
Comment (marianomerlo) [reply mule-packager/src/main/java/org/mule/tools/api/util/MavenProjectInformation.java:174]: 🤦

**Review by nicogon — APPROVED** [review]
LGTM, without test it's a bit hard to follow

File: mule-maven-plugin/pom.xml [hunk 4]
```diff
@@ -80,6 +86,14 @@
         </dependency>
     </dependencies>
 
+    <repositories>
+        <repository>
+            <id>mule-releases</id>
+            <name>Mule Release Repository</name>
+            <url>https://repository-master.mulesoft.org/nexus/content/repositories/releases</url>
+        </repository>
+    </repositories>
```
Comment (LucasLudueno) [inline mule-maven-plugin/pom.xml:95]: just to check it hasn't been added by accident

**Review by LucasLudueno — APPROVED** [review]

**Comment by marianomerlo** [comment]
```
[INFO] Mule Packager ...................................... SUCCESS [ 56.067 s]
[INFO] Mule Maven Plugin .................................. SUCCESS [ 19.860 s]
[INFO] Mule Maven Plugin Integration Tests ................ SUCCESS [  1.128 s]
[INFO] Mule Packager Integration Tests .................... SUCCESS [12:22 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  13:39 min
[INFO] Finished at: 2020-09-21T11:10:49-03:00
```

File: mule-maven-plugin/pom.xml (hunk 4 above)
Comment (marianomerlo) [reply mule-maven-plugin/pom.xml]: You were right, not needed.

**Review by nicogon — APPROVED** [review]
